meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx

```diff
@@ -356,6 +411,58 @@ export const SourceLayerItem = withTranslation()(
 			}
 		}
 
+		// TODO(Performance): use ResizeObserver to avoid style recalculations
+		// checkElementWidth = () => {
+		// 	if (this.state.itemElement && this._forceSizingRecheck) {
+		// 		this._forceSizingRecheck = false
+		// 		const width = getElementWidth(this.state.itemElement) || 0
+		// 		if (this.state.elementWidth !== width) {
+		// 			this.setState({
+		// 				elementWidth: width
+		// 			})
+		// 		}
+		// 	}
+		// }
+
+		private measureElement = () => {
+			if (!this.itemElement) return
+			const width = getElementWidth(this.itemElement) || 0
+			if (this.state.elementWidth !== width) {
+				this.setState({
+					elementWidth: width,
+				})
+			}
+		}
+
+		private onResize = (entries: ResizeObserverEntry[]) => {
+			const firstEntry = entries && entries[0]
+
+			if (firstEntry && firstEntry.contentRect && firstEntry.contentRect.width) {
+				const width = firstEntry.contentRect!.width
+				if (this.state.elementWidth !== width) {
+					this.setState({
+						elementWidth: width,
+					})
+				}
+			}
+		}
+
+		private mountResizeObserver() {
+			if (this.props.isLiveLine && !this._resizeObserver && this.itemElement) {
+				this._resizeObserver = new ResizeObserver(this.onResize)
+				this._resizeObserver.observe(this.itemElement)
```

Comment:
The `ResizeObserver` was refactored away because it added unnecessary overhead. We settled on using a different way of calculating this information - see https://github.com/nrkno/sofie-core/blob/a6f19b4d0644b5572280cdffc6cbb8f168b57d57/meteor/client/ui/SegmentTimeline/SourceLayerItem.tsx#L744

Reply:
@jstarpl I fixed this, but I feel like this function gets called too often now, please give another look